test/api/controllers/adminController.test.js

```diff
@@ -1,595 +1,510 @@
 var
-  should = require('should'),
-  _ = require('lodash'),
   Promise = require('bluebird'),
+  rewire = require('rewire'),
+  should = require('should'),
   sinon = require('sinon'),
-  params = require('rc')('kuzzle'),
-  Kuzzle = require.main.require('lib/api/Kuzzle'),
-  User = require.main.require('lib/api/core/models/security/user'),
-  Profile = require.main.require('lib/api/core/models/security/profile'),
-  Role = require.main.require('lib/api/core/models/security/role'),
+  AdminController = rewire('../../../lib/api/controllers/adminController'),
   RequestObject = require.main.require('kuzzle-common-objects').Models.requestObject,
   ResponseObject = require.main.require('kuzzle-common-objects').Models.responseObject,
   BadRequestError = require.main.require('kuzzle-common-objects').Errors.badRequestError,
-  PartialError = require.main.require('kuzzle-common-objects').Errors.partialError;
-
-require('sinon-as-promised')(Promise);
+  PartialError = require.main.require('kuzzle-common-objects').Errors.partialError,
+  sandbox = sinon.sandbox.create();
 
-describe('Test: admin controller', function () {
+describe('Test: admin controller', () => {
   var
+    adminController,
     kuzzle,
-    sandbox,
+    foo = {foo: 'bar'},
     index = '%text',
     collection = 'unit-test-adminController',
     requestObject;
 
-  before(() => {
-    kuzzle = new Kuzzle();
-
-    return kuzzle.start(params, {dummy: true});
-  });
-
   beforeEach(() => {
-    sandbox = sinon.sandbox.create();
+    kuzzle = {
```

Comment:
open question: maybe we could create a mock for Kuzzle instance within `test/mocks/`, with a default feature for each method of each internal component... and just override the methods we need within each test file ?